### PR TITLE
#570

### LINKED ISSUE #570: separate supplemental resources from required materials. (multiple assignees encouraged)
On one side, some apprentices think that they need to throughly look through every material, and are getting way behind because of it.  On the other, many well-meaning volunteers have thrown the kitchen sink into the "materials" section, when many or even all of them should be considered "supplemental resources" (supplemental meaning additional, extra, not required). 

Look through topic outlines, and change one so that:
- the "slides" are part of the "materials" section
- the "lesson" explicitly directs how to use the mandatory "materials" (ex: "Read through [these slides]() )
- optional materials are under a "supplemental resources" section.
- if you finish, or notice a topic outline looks fine, please check it off on this list or let a Techtonica admin know it's complete.

Topic outlines whose materials sections not been reviewed:
- [x] https://github.com/Techtonica/curriculum/blob/master/professionalism/professionalism.md
- [x] https://github.com/Techtonica/curriculum/blob/master/ergonomics/ergonomics.md

- [x] https://github.com/Techtonica/curriculum/blob/master/command-line/vim.md
 
- [x] https://github.com/Techtonica/curriculum/blob/master/html/html.md

- [x] https://github.com/Techtonica/curriculum/blob/master/git-version-control/gitignore.md
 
- [x] https://github.com/Techtonica/curriculum/blob/master/css/css.md

- [x] https://github.com/Techtonica/curriculum/blob/master/css/media-queries.md
 
- [x] https://github.com/Techtonica/curriculum/blob/master/ui-ux-design/ui-ux-design.md
- [x] https://github.com/Techtonica/curriculum/blob/master/web-patterns/web-patterns.md
- [x] https://github.com/Techtonica/curriculum/blob/master/bootstrap/bootstrap.md
- [x] https://github.com/Techtonica/curriculum/blob/master/professionalism/building-confidence.md

- [x] https://github.com/Techtonica/curriculum/blob/master/github-collaboration/github-collaboration.md
- [x] https://github.com/Techtonica/curriculum/blob/master/reading-and-writing-documentation/documentation.md
- [x] https://github.com/Techtonica/curriculum/blob/master/deploying/deploying-1-personal-site.md
 
- [x] https://github.com/Techtonica/curriculum/blob/master/javascript/javascript-1.md
- [x] https://github.com/Techtonica/curriculum/blob/master/javascript/javascript-2.md

- [x] https://github.com/Techtonica/curriculum/blob/master/runtime-complexity/runtime-complexity.md
- [x] https://github.com/Techtonica/curriculum/blob/master/testing-and-tdd/jasmine-testing.md
- [x] https://github.com/Techtonica/curriculum/blob/master/debugging/debugging.md
- [x] https://github.com/Techtonica/curriculum/blob/master/databases/relational-databases.md
- [x] https://github.com/Techtonica/curriculum/blob/master/react-js/react.md
- [x] https://github.com/Techtonica/curriculum/blob/master/data-structures/intro-to-data-structures.md
- [x] https://github.com/Techtonica/curriculum/blob/master/data-structures/stack.md
- [x] https://github.com/Techtonica/curriculum/blob/master/data-structures/deque.md

- [x] https://github.com/Techtonica/curriculum/blob/master/data-structures/trees.md
 
- [x] https://github.com/Techtonica/curriculum/blob/master/algorithms/intro-to-algorithms.md
- [x] https://github.com/Techtonica/curriculum/blob/master/algorithms/searching.md
- [x] https://github.com/Techtonica/curriculum/blob/master/algorithms/sorting.md

- [x] https://github.com/Techtonica/curriculum/blob/master/open-source/open-source.md
 
- [x] https://github.com/Techtonica/curriculum/blob/master/roles-in-tech/machine-learning.md

### PR DESCRIPTION
**Pull request by Rita-Okonkwo**
I created a supplemental resources section.
I also found out that in the "Materials and useful videos" section, one material named : "A deck of cards" isn't hyperlinked. Was this done on purpose?

### CODE REVIEW

**Comment by hoduchieu01** [comment]
Change the title of pull request and add link of issues in the description.

File: algorithms/sorting.md [hunk 1]
```diff
@@ -153,7 +156,7 @@ Well, we divide the list into lots of sublists. In fact, we divide them into lis
    1. Look at the first element of each, move smaller to the end of the result
    2. When one list empty, just copy the rest of the other list
 
-Let's review this video: https://www.youtube.com/watch?v=KF2j-9iSf4Q
+Let's review this video: [Watch this video, it is also available in the materials section)https://www.youtube.com/watch?v=KF2j-9iSf4Q
```
Comment (hoduchieu01) [inline algorithms/sorting.md]: This line has some errors about markdown syntax.

Comment (manaswinidas) [inline algorithms/sorting.md]: Follow this: https://www.markdownguide.org/cheat-sheet/

**Comment by manaswinidas** [comment]
@Rita-Okonkwo Please give the issue number in the description so that the issue is automatically referenced.

**Comment by manaswinidas** [comment]
Fixes #570

**Comment by Rita-Okonkwo** [comment]
Okay thanks alot @hoduchieu01 @manaswinidas I've made the changes now

**Comment by Rita-Okonkwo** [comment]
Okay thanks alot @hoduchieu01 @manaswinidas I've made the changes now

File: algorithms/sorting.md [hunk 2]
```diff
@@ -23,8 +23,11 @@
 - [Bubble Sort With Hungarian Folk Dance](https://www.youtube.com/watch?v=lyZQPjUT5B4)
 - [Merge Sort With German Folk Dance](https://www.youtube.com/watch?v=XaqR3G_NVoo)
 - A Deck of Cards
-- [Sorting Algorithms Conceptually](https://code.likeagirl.io/sorting-algorithms-conceptually-2bfbb3968388)
 
+### Supplemental Resources
+***These materials are optional, that is, they are not required.***
```
Comment (alodahl) [inline algorithms/sorting.md]: Please remove line 28, and move the supplemental resources section to be last, after check for understanding. After that, I’ll merge 😊

Comment (Rita-Okonkwo) [reply algorithms/sorting.md]: Alright @alodahl I've done that, thanks

**Comment by alodahl** [comment]
Closes #570

**Pull request merged**